### DISCUSSION TITLE
Help with `BrokenProcessPool` error

### DISCUSSION THREAD

**Discussion by avaudano** [HALFpipe/HALFpipe#133] [Q&A] [answered]
Dear all, 
i am new using halfpipe and I have a probably naive question. 
I need to run halfpipe pipeline to analysis the resting state analysis of 90 subjects. 
The analysis crashed and the error message was "BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending." I
I attach the err.txt file. 
Is anyone who might help me?
Thank-you very much, 

Anna[
[err.txt](https://github.com/HALFpipe/HALFpipe/files/6676864/err.txt)
](url)

**Comment by HippocampusGirl** [comment, marked as answer]
Hi Anna,
This looks like a memory issue. The `BrokenProcessPool` message usually means that the operating system stopped a process that used too much memory (RAM), thus breaking the pool of processes that HALFpipe was using to run commands. Could you try adding `--subject-chunks` to the end of the command you use to start HALFpipe? It makes processing a bit slower but should significantly reduce memory usage. If that still doesn't work, could you try adding `--nipype-n-procs 1` to the end?
I hope to improve memory usage in the next release. Please see https://github.com/HALFpipe/HALFpipe/issues/125 to keep up to date with those efforts :-)

**Reply by avaudano** [reply]
Thank-you very much!
